accepted/2021/InboxSourceGenerators.md

```diff
@@ -0,0 +1,165 @@
+
+## Scenarios and User Experience
+
+A developer can use source generators for framework API without making any changes to their project.  Creating a new project should be sufficient.
+
+A developer can use source generators for components in NuGet packages by referencing the NuGet package for that component.  
+
+When a component ships both in a NuGet package and a framework, the developer has a consistent experience with the source generator as they do with the library itself.  If the package referenced is newer than the framework the source generator from the package will be used as is already the case for the library in that package.  Similarly if the framework is newer than the NuGet package the framework's source generator will be used as is already the case for the library in the framework.
+
+A developer can receive updates to a source generator through normal channels which they already obtain updates today.
+
+A developer can update their version of Visual Studio or Dotnet SDK and have confidence that new versions of the toolchain won't impact their applications which target older frameworks, outside of known and published servicing updates.
+
+A developer has confidence that their library using inbox source generators will continue to work in the same way a library that doesn't use inbox source generators works on future versions of .NET.  Source gener
+
+## Requirements
+
+### Goals
+
+- Enable consumption of source generators for components in shared frameworks
+- Address conflict resolution between source generators in shared frameworks and NuGet packages
+- Define servicing process for shared framework source generators
+- Establish guidelines building and versioning shared framework source generators
+- Describe compatibility requirements for source generators.
+
+### Non-Goals
+
+- Define a process for components not in shared frameworks to contribute source generators.
+- Define any changes to the source generator architecture.
+- Define any implementation details of source generators or source generator tests.
+- Define a process around internal or non-shipping source generators
+
+## Stakeholders and Reviewers
+
+- ASP.NET - also building shared framework source generators.
+- .NET Libraries team - building multiple shared framework source generators.
+- Mono team - interested in providing source generators.
+- .NET Runtime team - interested in providing source generators.
+- Roslyn team - responsible for source generator infrastructure.
+- .NET SDK team - responsible for resolving shared frameworks and doing conflict resolution with content.
+- NuGet team - FYI only, no changes requested
+
+## Design
```

Comment:
I think we should mention what this design means for branches and the servicing process. Specifically, that if a bug is found in a generator we may need to make the same fix in multiple branches (potentially using different code), depending on how many frameworks we support and in how many frameworks the bug is present. In other words, the process would be virtually identical to how we service runtime code.

Reply:
I'll add to the servicing section.